routingconn.go

```diff
@@ -80,6 +87,17 @@ func dialRoutingConn(ctx context.Context, address string, opts *routingConnOptio
 		dialOpts = append(dialOpts, perRpcDialOpt)
 	}
 
+	clientOpts := []otelgrpc.Option{
+		otelgrpc.WithPropagators(propagation.TraceContext{}),
+	}
+	if opts.TracerProvider != nil {
+		clientOpts = append(clientOpts, otelgrpc.WithTracerProvider(opts.TracerProvider))
+	}
+	if opts.MeterProvider != nil {
+		clientOpts = append(clientOpts, otelgrpc.WithMeterProvider(opts.MeterProvider))
+	}
+	dialOpts = append(dialOpts, grpc.WithStatsHandler(otelgrpc.NewClientHandler(clientOpts...)))
```

Comment:
Should this only be set if there is a metricsProvider configured?

Reply:
Hmm maybe not actually

Reply:
Yes it should, StatsHandler also emcompasses metrics.